server/sonar-server/src/main/java/org/sonar/server/webhook/ws/CreateAction.java

```diff
@@ -0,0 +1,211 @@
+import static org.sonar.server.webhook.ws.WebhooksWsParameters.NAME_PARAM;
+import static org.sonar.server.webhook.ws.WebhooksWsParameters.NAME_PARAM_MAXIMUM_LENGTH;
+import static org.sonar.server.webhook.ws.WebhooksWsParameters.ORGANIZATION_KEY_PARAM;
+import static org.sonar.server.webhook.ws.WebhooksWsParameters.ORGANIZATION_KEY_PARAM_MAXIMUM_LENGTH;
+import static org.sonar.server.webhook.ws.WebhooksWsParameters.PROJECT_KEY_PARAM;
+import static org.sonar.server.webhook.ws.WebhooksWsParameters.PROJECT_KEY_PARAM_MAXIMUN_LENGTH;
+import static org.sonar.server.webhook.ws.WebhooksWsParameters.URL_PARAM;
+import static org.sonar.server.webhook.ws.WebhooksWsParameters.URL_PARAM_MAXIMUM_LENGTH;
+import static org.sonar.server.ws.KeyExamples.KEY_ORG_EXAMPLE_001;
+import static org.sonar.server.ws.KeyExamples.KEY_PROJECT_EXAMPLE_001;
+import static org.sonar.server.ws.KeyExamples.NAME_WEBHOOK_EXAMPLE_001;
+import static org.sonar.server.ws.KeyExamples.URL_WEBHOOK_EXAMPLE_001;
+import static org.sonar.server.ws.WsUtils.checkFoundWithOptional;
+import static org.sonar.server.ws.WsUtils.checkStateWithOptional;
+import static org.sonar.server.ws.WsUtils.writeProtobuf;
+import static org.sonarqube.ws.Webhooks.CreateWsResponse.Webhook;
+import static org.sonarqube.ws.Webhooks.CreateWsResponse.newBuilder;
+
+public class CreateAction implements WebhooksWsAction {
+
+  private static final int MAX_NUMBER_OF_WEBHOOKS = 10;
+
+  private final DbClient dbClient;
+  private final UserSession userSession;
+  private final DefaultOrganizationProvider defaultOrganizationProvider;
+  private final UuidFactory uuidFactory;
+  private final WebhookSupport webhookSupport;
+
+  public CreateAction(DbClient dbClient, UserSession userSession, DefaultOrganizationProvider defaultOrganizationProvider,
+                      UuidFactory uuidFactory, WebhookSupport webhookSupport) {
+    this.dbClient = dbClient;
+    this.userSession = userSession;
+    this.defaultOrganizationProvider = defaultOrganizationProvider;
+    this.uuidFactory = uuidFactory;
+    this.webhookSupport = webhookSupport;
+  }
+
+  @Override
+  public void define(WebService.NewController controller) {
+
+    WebService.NewAction action = controller.createAction(ACTION_CREATE)
+      .setPost(true)
+      .setDescription("Create a Webhook.<br>" +
+        "Requires the global, organization or project permission.")
+      .setSince("7.1")
+      .setResponseExample(getClass().getResource("example-webhook-create.json"))
+      .setHandler(this);
+
+    action.createParam(NAME_PARAM)
+      .setRequired(true)
+      .setMaximumLength(NAME_PARAM_MAXIMUM_LENGTH)
+      .setDescription("The name of the webhook to create")
+      .setExampleValue(NAME_WEBHOOK_EXAMPLE_001);
+
+    action.createParam(URL_PARAM)
+      .setRequired(true)
+      .setMaximumLength(URL_PARAM_MAXIMUM_LENGTH)
+      .setDescription("The url to be called by the webhook")
```

Comment:
The previous description should be kept: "Server endpoint that will receive the webhook payload, for example 'http://my_server/foo'. If HTTP Basic authentication is used, HTTPS is recommended to avoid man in the middle attacks. Example: 'https://myLogin:myPassword@my_server/foo'"

Reply:
done